app/models/katello/pool.rb

```diff
@@ -50,13 +50,16 @@ class Pool < Katello::Model
 
     validates_lengths_from_database
 
-    validates :subscription, :presence => true
+    validates :subscription, :presence => { :message => N_("cannot be blank. If this is during a manifest import or refresh, try running foreman-rake katello:upgrades:4.8:fix_incorrect_providers.") }
```

Comment:
Given that:

- the upgrade rake task has been run already when this code hits via installer (correcting any incorrect providers)
- you have put in place validations that prevent mismatched providers (should give validation errors of their own)
- you no longer import orphaned products

=> This validation should never, ever fire since it's not based on any kind of user input. And if it does fire, I don't think it would be because of anything to do w/ the providers at that point but some other bug the user can't work around themselves.

The upgrade rake tasks are also periodically removed, so the message would direct the user to do something totally un-doable in a few releases

Reply:
okay, good points. :P  Removing.